commit 250077dfedf478e242088dba85578158b33236bc (refs/changes/67/2367/1)
Author: Angela Fong <angela.fong@joyent.com>
Date:   2017-08-09T21:28:30-07:00 (2 years, 2 months ago)
    
    ADMINUI-2384 AdminUI should handle API calls with undefined route name

diff --git a/lib/adminui.js b/lib/adminui.js
index e66c2e1c..768ab268 100644
--- a/lib/adminui.js
+++ b/lib/adminui.js
@@ -903,7 +903,7 @@ ADMINUI.prototype.createServer = function () {
         }
         var method = req.method;
         var body =
-            !(method === 'POST' && route.name === 'authenticate') &&
+            !(method === 'POST' && route && route.name === 'authenticate') &&
             !(method === 'GET' && Math.floor(res.statusCode/100) === 2);
         auditLogger({
             log: req.log.child({ component: 'audit', route: route && route.name }, true),
